flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -986,6 +987,20 @@ public TableResultInternal executeInternal(Operation operation) {
                     for (CatalogPartitionSpec spec : dropPartitionsOperation.getPartitionSpecs()) {
                         catalog.dropPartition(tablePath, spec, ifExists);
                     }
+                } else if (alterTableOperation instanceof PartitionRenameOperation) {
+                    PartitionRenameOperation partitionRenameOperation =
+                            (PartitionRenameOperation) alterTableOperation;
+                    ObjectPath tablePath =
+                            partitionRenameOperation.getTableIdentifier().toObjectPath();
+                    // drop old partition first, require old partition exist
+                    catalog.dropPartition(
```

Comment:
For `PartitionRenameOperation`, I suggest to make catalog itself to deal with it insteading always drop partition +  add partition. 
For example, Hive provides renamePartition API. Although it also first drops and then adds partition, but it does more things such like update added partition to refer to the old partition's location.